### ISSUE TITLE
Can't create a new project

### ISSUE THREAD

**Issue by WouterJ** [symfony/symfony-installer#46]
Whatever I do, I always get:

    $>php symfony.phar new blog

     Downloading Symfony...
       4.74 MB/4.74 MB ============================================================  100%
    
     Preparing project...
    
      [RuntimeException]
      Symfony can't be installed because the downloaded package is corrupted
      or because the installer doesn't have enough permissions to uncompress and
      rename the package contents.
      To solve this issue, try installing Symfony again and check the permissions of
      the C:\Users\Wouter\Documents\web\sf-world directory
    
    new name [version]

I'm using Windows 8.1.

**Comment by Putr** [comment]
I'm using version 0.5.0 and I still have this problem (Kubuntu 14.04, PHP 5.5).

I also get this error:
```
  [Symfony\Component\Filesystem\Exception\IOException]
  Failed to write file "[path-to-project]/.142013509154a58ab3d03f3/symfony.tgz".
```

**Comment by fredpeaks** [comment]
Sorry but I have exactly the same error message : "Symfony can't be installed because the downloaded package is corrupted or because the installer doesn't have enough permissions to uncompress and rename the package contents...".
I'm on Windows 8.

**Comment by xabbuh** [comment]
@fredpeaks Which version of the installer do you use?

**Comment by fredpeaks** [comment]
@xabbuh I use the version which can be downloaded at http://symfony.com/installer (today) :
`Symfony Installer version 0.5.0`
I followed the "[Official Symfony Best Pratices](http://symfony.com/doc/current/best_practices/creating-the-project.html#windows-systems)" guide here to download it with this command:
`php -r "readfile('http://symfony.com/installer');" > symfony.phar`

**Comment by xabbuh** [comment]
Do you get some more messages when you run the command with the `-vvv` option?

**Comment by javiereguiluz** [comment]
@fredpeaks when things go wrong, the PHAR version is not the best way to detect what's happening. I recomend you to download the latest version of the installer from GitHub and then try to create a Symfony project. You only have to do this:

```
c:\> mkdir installer
c:\> cd installer
c:\installer\> git clone https://github.com/symfony/symfony-installer.git
c:\installer\> composer install
c:\installer\> php ./symfony new project
```

**Comment by fredpeaks** [comment]
@xabbuh Here is the verbose output for the "symfony.phar new" command:
```
[RuntimeException]
Symfony can't be installed because the downloaded package is corrupted
or because the installer doesn't have enough permissions to uncompress and
rename the package contents.
To solve this issue, check the permissions of the $ %s directory and
try installing Symfony again.

Exception trace:
 () at phar://L:/symfony.phar/src/Symfony/Installer/NewCommand.php:297
 Symfony\Installer\NewCommand->extract() at phar://L:/symfony.phar/src/Symfony/Installer/NewCommand.php:68
 Symfony\Installer\NewCommand->execute() at phar://L:/symfony.phar/vendor/symfony/console/Symfony/Component/Console/Command/Command.php:252
 Symfony\Component\Console\Command\Command->run() at phar://L:/symfony.phar/vendor/symfony/console/Symfony/Component/Console/Application.php:874
 Symfony\Component\Console\Application->doRunCommand() at phar://L:/symfony.phar/vendor/symfony/console/Symfony/Component/Console/Application.php:195
 Symfony\Component\Console\Application->doRun() at phar://L:/symfony.phar/vendor/symfony/console/Symfony/Component/Console/Application.php:126
 Symfony\Component\Console\Application->run() at phar://L:/symfony.phar/symfony:33
 require() at L:\symfony.phar:10
```

@javiereguiluz When I try with your method, directly with source code and no phar, it's working correctly, no error.

Another thing : the same phar is working fine on other PCs with Windows 8.

**Comment by Pierstoval** [comment]
With another month of commits and the recent `1.0.0` tag, do you still have this issue after updating the installer ?

**Comment by hexmode** [comment]
Works for me with separate /home and /tmp and / partitions on debian

**Comment by georaldc** [comment]
Same problem here, with the same exception trace as @fredpeaks.

Manual install using composer works fine

**Comment by Pierstoval** [comment]
Now that there is a new release, can ya'll upgrade the installer and test again with the `-vvv` argument for we can check the potential new backtrace?

**Comment by leto3087** [comment]
I have tested with the last version and still the same error on Windows 8 (I've cloned the repository and not used the .phar).
Here is the verbose output:

```
  [RuntimeException]
  Symfony can't be installed because the downloaded package is corrupted
  or because the installer doesn't have enough permissions to uncompress and
  rename the package contents.
  To solve this issue, check the permissions of the L:\ directory and
  try executing this command again:
  sf-installer/symfony new symfonytest



Exception trace:
 () at L:\sf-installer\src\Symfony\Installer\DownloadCommand.php:208
 Symfony\Installer\DownloadCommand->extract() at L:\sf-installer\src\Symfony\Installer\NewCommand.php:59
 Symfony\Installer\NewCommand->execute() at L:\sf-installer\vendor\symfony\console\Symfony\Component\Console\Command\Command.php:252
 Symfony\Component\Console\Command\Command->run() at L:\sf-installer\vendor\symfony\console\Symfony\Component\Console\Application.php:874
 Symfony\Component\Console\Application->doRunCommand() at L:\sf-installer\vendor\symfony\console\Symfony\Component\Console\Application.php:195
 Symfony\Component\Console\Application->doRun() at L:\sf-installer\vendor\symfony\console\Symfony\Component\Console\Application.php:126
 Symfony\Component\Console\Application->run() at L:\sf-installer\symfony:36





  [UnexpectedValueException]
  FilesystemIterator::__construct(C:\Users\chris\AppData\Local\Temp\1428068430551e984e5dc71\,C:\Users\chris\AppData\Local\Temp\1428068430551e984e5dc71\): Le
  fichier sp?cifi? est introuvable. (code: 2)



Exception trace:
 () at L:\sf-installer\vendor\raulfraile\distill\src\Distill.php:176
 FilesystemIterator->__construct() at L:\sf-installer\vendor\raulfraile\distill\src\Distill.php:176
 Distill\Distill->extractWithoutRootDirectory() at L:\sf-installer\src\Symfony\Installer\DownloadCommand.php:186
 Symfony\Installer\DownloadCommand->extract() at L:\sf-installer\src\Symfony\Installer\NewCommand.php:59
 Symfony\Installer\NewCommand->execute() at L:\sf-installer\vendor\symfony\console\Symfony\Component\Console\Command\Command.php:252
 Symfony\Component\Console\Command\Command->run() at L:\sf-installer\vendor\symfony\console\Symfony\Component\Console\Application.php:874
 Symfony\Component\Console\Application->doRunCommand() at L:\sf-installer\vendor\symfony\console\Symfony\Component\Console\Application.php:195
 Symfony\Component\Console\Application->doRun() at L:\sf-installer\vendor\symfony\console\Symfony\Component\Console\Application.php:126
 Symfony\Component\Console\Application->run() at L:\sf-installer\symfony:36


new directory [version]

```

**Comment by WouterJ** [comment]
@leto3087 can you please run `composer update` in `L:\sf-installer`? Seems you're using Distill 0.9.2, while 0.9.8 is the latest (not sure if that's going to make a difference).

Could you maybe add a var_dump on the `$this->extract()` call in `Distill#extractWithoutRootDirectory`? (https://github.com/raulfraile/distill/blob/v0.9.3/src/Distill.php#L190 ). I have the feeling that they archive isn't extracted correctly.

**Comment by leto3087** [comment]
@WouterJ After a composer update the error is still the same.
It seems that the problem is related to targz. When I manually remove the "tgz" extension form the available choices (https://github.com/symfony/symfony-installer/blob/master/src/Symfony/Installer/DownloadCommand.php#L79), it's working fine.
When using tgz, I can see that the PharData extractor is used which seems weird to me.

**Comment by xabbuh** [comment]
@leto3087 This looks like an issue in distill to me. Can you open an issue over there?

**Comment by WouterJ** [comment]
PharData can perfectly extract tgz sources (it works on my PC too): http://php.net/manual/en/phardata.extractto.php

**Comment by leto3087** [comment]
After some research, it seems that the issue is related to the $file path in PharData::extract() which in my case begin with "L:\" because my sites are at the root of my L: drive.
And PharData constructor detect this path as a URL.

To resolve it, the creation of the downloadedFilePath in DownloadCommand.php (https://github.com/symfony/symfony-installer/blob/master/src/Symfony/Installer/DownloadCommand.php#L120) must be changed to avoid path like this : "'L:\\folder".

**Comment by sstok** [comment]
What if the path is changed to use a forward slash instead backslash? or prefix with 'file://' to indicate its a local file-path.

**Comment by Pierstoval** [comment]
It would certainly still treat it as an url, because "file://" is kind of a local url ... :confused:

**Comment by leto3087** [comment]
@sstok @Pierstoval 
Sorry i realised that the path displayed in my previous comment was "L:\".
But the problem is that the path begin with : "L:\\\".
I've edited my previous comment !

**Comment by Pierstoval** [comment]
Can you try, while debugging, to change this value to "L:/" ?

**Comment by leto3087** [comment]
@Pierstoval This is exactly what I've tried to explain : it's working when the path begin with "L:\" (and not "L:\\\").
The getcwd() call (https://github.com/symfony/symfony-installer/blob/master/src/Symfony/Installer/DownloadCommand.php#L120) returns  "L:\".
So I think the code should be modified like this to work in every case :
```
$this->downloadedFilePath = rtrim(getcwd(), DIRECTORY_SEPARATOR).DIRECTORY_SEPARATOR.'.'.uniqid(time()).DIRECTORY_SEPARATOR.'symfony.'.pathinfo($symfonyArchiveFile, PATHINFO_EXTENSION);
```

**Comment by Pierstoval** [comment]
> ```php
$this->downloadedFilePath =
    rtrim(getcwd(), DIRECTORY_SEPARATOR).
    DIRECTORY_SEPARATOR.
    '.'.uniqid(time()).
    DIRECTORY_SEPARATOR.
    'symfony.'.
    pathinfo($symfonyArchiveFile, PATHINFO_EXTENSION);
```

:+1: for this update if it works for every environment

**Comment by xabbuh** [comment]
UNIX/Linux-based systems as well as Mac OS X shouldn't have any issues with this.

@leto3087 Do you want to create a pull request for this?

**Comment by leto3087** [comment]
@xabbuh Here is the new pull request: https://github.com/symfony/symfony-installer/pull/140
:)

**Comment by avinfo79** [comment]
NOT only Windows OS!
In VirtualBox runing UBUNTU 12.04.LTS with vbox_sf shared folder 

/media/sf_/host/Symfony2 -  link folder to /var/www/host/Symfony2

/var/www/host/Symfony2$ sudo symfony new myproject
```
 [RuntimeException]                                                                        
  Symfony can't be installed because the downloaded package is corrupted                    
  or because the installer doesn't have enough permissions to uncompress and                
  rename the package contents.                                                              
  To solve this issue, check the permissions of the /media/sf_/host/Symfony2 directory and  
  try executing this command again:                                                         
  symfony new myproject                   

Exception trace:
 () at phar:///usr/local/bin/symfony/src/Symfony/Installer/DownloadCommand.php:208
 Symfony\Installer\DownloadCommand->extract() at phar:///usr/local/bin/symfony/src/Symfony/Installer/NewCommand.php:59
 Symfony\Installer\NewCommand->execute() at phar:///usr/local/bin/symfony/vendor/symfony/console/Symfony/Component/Console/Command/Command.php:252
 Symfony\Component\Console\Command\Command->run() at phar:///usr/local/bin/symfony/vendor/symfony/console/Symfony/Component/Console/Application.php:874
 Symfony\Component\Console\Application->doRunCommand() at phar:///usr/local/bin/symfony/vendor/symfony/console/Symfony/Component/Console/Application.php:195
 Symfony\Component\Console\Application->doRun() at phar:///usr/local/bin/symfony/vendor/symfony/console/Symfony/Component/Console/Application.php:126
 Symfony\Component\Console\Application->run() at phar:///usr/local/bin/symfony/symfony:35
 require() at /usr/local/bin/symfony:10
                                                                                                                                  
  [RuntimeException]                                                                                                               
  Could not delete /tmp/142877890155296f95b3417/Symfony/web/bundles/acmedemo: rmdir(/tmp/142877890155296f95b3417/Symfony/web/bund  
  les/acmedemo): Not a directory                                                                                                   
                                                                                                                                   
Exception trace:
 () at phar:///usr/local/bin/symfony/vendor/raulfraile/distill/src/Extractor/Util/Filesystem.php:177
 Distill\Extractor\Util\Filesystem->rmdir() at phar:///usr/local/bin/symfony/vendor/raulfraile/distill/src/Extractor/Util/Filesystem.php:107
 Distill\Extractor\Util\Filesystem->removeDirectoryPhp() at phar:///usr/local/bin/symfony/vendor/raulfraile/distill/src/Extractor/Util/Filesystem.php:215
 Distill\Extractor\Util\Filesystem->copyThenRemove() at phar:///usr/local/bin/symfony/vendor/raulfraile/distill/src/Extractor/Util/Filesystem.php:255
 Distill\Extractor\Util\Filesystem->rename() at phar:///usr/local/bin/symfony/vendor/raulfraile/distill/src/Distill.php:207
 Distill\Distill->extractWithoutRootDirectory() at phar:///usr/local/bin/symfony/src/Symfony/Installer/DownloadCommand.php:186
 Symfony\Installer\DownloadCommand->extract() at phar:///usr/local/bin/symfony/src/Symfony/Installer/NewCommand.php:59
 Symfony\Installer\NewCommand->execute() at phar:///usr/local/bin/symfony/vendor/symfony/console/Symfony/Component/Console/Command/Command.php:252
 Symfony\Component\Console\Command\Command->run() at phar:///usr/local/bin/symfony/vendor/symfony/console/Symfony/Component/Console/Application.php:874
 Symfony\Component\Console\Application->doRunCommand() at phar:///usr/local/bin/symfony/vendor/symfony/console/Symfony/Component/Console/Application.php:195
 Symfony\Component\Console\Application->doRun() at phar:///usr/local/bin/symfony/vendor/symfony/console/Symfony/Component/Console/Application.php:126
 Symfony\Component\Console\Application->run() at phar:///usr/local/bin/symfony/symfony:35
 require() at /usr/local/bin/symfony:10
```

**Comment by javiereguiluz** [comment]
I'm closing this issue because the error has been fixed as of https://github.com/symfony/symfony-installer/pull/140 . Besides, this ultra long thread has become very hard to follow it. If you suffer other errors, please create new issues. Thanks!
